utilities/export-template.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+""" This is a generic file exporter for NIPAP
+
+    It fetches prefixes based on a provided query string and feeds that into a
+    template that can be used to produce a configuration file for another
+    program, such as ISC DHCP.
+
+    It will read .nipaprc per default or rely on command line options for
+    connection settings to the backend.
+"""
+
+from __future__ import print_function
+
+import ConfigParser
+import os
+# use local pynipap, useful if we are developing
+import sys
+sys.path.append('../pynipap')
+from pynipap import Prefix, Pool, VRF
+import pynipap
+
+import IPy
+import jinja2
+
+#
+# Fill in your code here
+#
+
+class ConfigExport:
+    def __init__(self, template_filename, output_filename):
+        """ Init!
+        """
+        self.output_filename = output_filename
+
+        env = jinja2.Environment(loader=jinja2.FileSystemLoader(['./']))
+        self.t = env.get_template(template_filename)
+        self.prefixes = []
+
+
+    def get_prefixes(self, query):
+        """ Get prefix data from NIPAP
+        """
+        try:
+            res = Prefix.smart_search(query, {})
+        except socket.error:
+            print >> sys.stderr, "Connection refused, please check hostname & port"
+            sys.exit(1)
+        except xmlrpclib.ProtocolError:
+            print >> sys.stderr, "Authentication failed, please check your username / password"
+            sys.exit(1)
+
+        for p in res['result']:
+            p.prefix_ipy = IPy.IP(p.prefix)
+            self.prefixes.append(p)
+
+
+    def write_conf(self):
+        """ Write the config to file
+        """
+        f = open(self.output_filename, 'w')
+        print(self.t.render(prefixes=self.prefixes), file=f)
+        f.close()
+
+
+
+if __name__ == '__main__':
+    # read configuration
+    cfg = ConfigParser.ConfigParser()
+    cfg.read(os.path.expanduser('~/.nipaprc'))
+
+    import optparse
+    parser = optparse.OptionParser()
+    # standard options to specify nipapd connection
+    parser.add_option('--username', default='', help="Username")
+    parser.add_option('--password', default='', help="Password")
+    parser.add_option('--host', help="NIPAP backend host")
+    parser.add_option('--port', default=1337, help="NIPAP backend port")
```

Comment:
The default-argument should probably be removed here as it leads to the port never being taken from ~/.nipaprc. As ```options.port``` is always set due to the default-value, ```options.port or cfg.get('global', 'port')``` at line 90 will never yield the value from cfg. Originally an issue with boilerplate.py, I'll create an issue for that as well.